### PR TITLE
Open UDP/4800 ports between master and worker nodes on AWS

### PR DESCRIPTION
**Pull request by sridhargaddam**
Recently Submariner pods are configured as infrastructure pods
which will ignore any taints. Because of this change, they will
get scheduled even on the Master nodes. To validate this use-case
Submariner e2e test pods are also modified to tolerate any taints.

So, we should open UDP/4800 ports between the master and worker
nodes in an AWS cluster. This PR handles this change in terraform
scripts.

Signed-Off-by: Sridhar Gaddam <sgaddam@redhat.com>

### CODE REVIEW

**Review by nyechiel — APPROVED** [review]
LGTM, thanks @sridhargaddam

**Review by dfarrell07 — APPROVED** [review]

File: tools/openshift/ocp-ipi-aws/ocp-ipi-aws-prep/ec2-resources.tf [hunk 1]
```diff
@@ -23,6 +39,27 @@ resource "aws_security_group_rule" "worker_sg_vxlan_rule" {
   type                     = "ingress"
 }
 
+# Add a rule for vxlan traffic from master nodes to worker nodes.
```
Comment (mangelajo) [inline tools/openshift/ocp-ipi-aws/ocp-ipi-aws-prep/ec2-resources.tf:42]: We also need to add master node to master node, like here:

https://github.com/submariner-io/submariner/blob/master/tools/openshift/ocp-ipi-aws/ocp-ipi-aws-prep/ec2-resources.tf#L17

**Review by mangelajo — CHANGES_REQUESTED** [review]

File: tools/openshift/ocp-ipi-aws/ocp-ipi-aws-prep/ec2-resources.tf (hunk 1 above)
Comment (mangelajo) [reply tools/openshift/ocp-ipi-aws/ocp-ipi-aws-prep/ec2-resources.tf:42]: Oh, as talked in slack, in the existing IPI topology, the gateway node won't ever be part of the master SG.
and 4800 is used for node to GW communication.

**Review by mangelajo — APPROVED** [review]

**Review by manosnoam — COMMENTED** [review]
A general question - isn't this fix relevant only for E2E execution, but not for Submariner functionality ? 
Shouldn't we fix it on the E2E test flow, instead of in ocp-ipi-aws ?

**Comment by sridhargaddam** [comment]
> A general question - isn't this fix relevant only for E2E execution, but not for Submariner functionality ?
> Shouldn't we fix it on the E2E test flow, instead of in ocp-ipi-aws ?

The problem was that our e2e pods are now tolerating all taints (i.e., they will even get scheduled on Master nodes when we run e2e tests). So we should open the necessary UDP/4800 ports between Master/worker nodes. We don't need any additional changes to our e2e test flow.

**Pull request merged**